### PR TITLE
Updating Xcode version to 12.5.

### PR DESCRIPTION
**Pull request by rdeassis**
### Platforms Impacted
- [x] iOS
- [x] macOS

### Description of changes

Xcode 12.5 is available in the GitHub images now.
This change updates the pipelines to use it according to the environment info listed on the links below:

- https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11.0-Readme.md
- https://github.com/actions/virtual-environments

### Verification

Local build using Big Sur (11.3.1) with Xcode 12.5.

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/560)

### CODE REVIEW

**Review by Saadnajmi — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Fluent UI Apple contains native UIKit and AppKit controls aligned with [Microsof
 #### Requirements
 
 - iOS 13+ or macOS 10.14+
-- Xcode 11+
+- Xcode 12.5+
```
Comment (harrieshin) [inline README.md:19]: while our pipeline is updating... it doesn't mean the sdk requires 12.5 right?

File: .ado/templates/fluentui-apple-publish-nuget-job.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 jobs:
 - job: fluentui_apple_publish_nuget
   pool:
-    vmImage: 'macos-10.15'
+    vmImage: 'macos-11.0'
```
Comment (harrieshin) [inline .ado/templates/fluentui-apple-publish-nuget-job.yml:4]: curious if this works

Comment (rdeassis) [reply .ado/templates/fluentui-apple-publish-nuget-job.yml:4]: The OS selection seems to be working as per the log below.
But it looks like #486 changes made the work flow stop using our script to select the correct Xcode version.

Screen Shot 2021-05-10 at 3 18 57 PM

File: README.md (hunk 1 above)
Comment (rdeassis) [reply README.md:19]: I agree that it's not a requirement that we use Xcode 12.5, but we also don't validate previous versions once we're running on 12.5.

I can revert it or put the Xcode version in a "recommended" section instead?

Comment (harrieshin) [reply README.md:19]: I think having requires 11 makes sense for iOS 13+ requirement. yeah, maybe recommended latest Xcode might be something we could add?

**Pull request closed without merging**